ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1292,6 +1299,15 @@ void checkAndMaybeComplete() {
         }
       }
       if (done) {
+        // the chunk is complete now. We can charge against quota for the chunk.
+        if (quotaChargeEventListener != null) {
+          try {
+            quotaChargeEventListener.onQuotaChargeEvent();
+          } catch (RouterException rEx) {
+            // For now we only log for quota charge exceptions for in progress requests.
+            logger.info("Exception {} while handling quota charge event", rEx.toString());
+          }
+        }
```

Comment:
We have to set the exception fo this PutOperation here in the catch statement, otherwise, it's not consistent with GetBlobOperation since we are setting exception in GetBlobOperation.

Reply:
Removed setting exception in GetBlobOperation as we are not throttling for in-progress requests for now.